py/desiutil/bitmask.py

```diff
@@ -100,6 +100,52 @@ def __str__(self):
     # def __repr__(self):
     #     return "_MaskBit(name='{0.name}', bitnum={0.bitnum:d}, comment='{0.comment}')".format(self)
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        """When working with Numpy functions and objects, replace the input
+        ``_MaskBit`` object with its ``.mask`` attribute.
+        """
+        #
+        # Filter inputs and replace any _MaskBit with a Python int.
+        #
+        args = []
+        for input_ in inputs:
+            if isinstance(input_, _MaskBit):
+                args.append(input_.mask)
+            else:
+                args.append(input_)
+        #
+        # This is boilerplate designed to capture ``out`` keywords.
+        #
+        # try:
+        #     outputs = kwargs['out']
+        # except KeyError:
+        #     outputs = (None,) * ufunc.nout
+        #
+        # Identify another argument that we can pass the computation to.
+        #
+        sup = None
+        for arg in args:
+            if hasattr(arg, '__array_ufunc__'):
+                sup = arg
+
+        if sup is None:
+            return NotImplemented
+
+        return sup.__array_ufunc__(ufunc, method, *args, **kwargs)
+        # results = sup.__array_ufunc__(ufunc, method, *args, **kwargs)
+        # if results is NotImplemented:
+        #     return NotImplemented
+        #
+        # Deal with multiple outputs.
+        #
+        # if ufunc.nout == 1:
+        #     results = (results,)
+
+        # results = tuple((np.asarray(result) if output is None else output)
+        #                 for result, output in zip(results, outputs))
+
+        # return results[0] if len(results) == 1 else results
```

Comment:
When we've converged on what is necessary, let's remove the remaining commented out boilerplate before merging.

Reply:
Clean up done.